src/probnum/statespace/sde.py

```diff
@@ -231,9 +235,58 @@ def _solve_mde_forward(self, rv, t, dt, _diffusion=1.0):
             "sol_cov": sol_cov,
         }
 
+    def _solve_mde_forward_sqrt(self, rv, t, dt, _diffusion=1.0):
+        """Solve forward moment differential equations (MDEs) using a square-root
+        implementation."""
+        mde, y0 = self._setup_vectorized_mde_forward_sqrt(
+            rv,
+            _diffusion=_diffusion,
+        )
+
+        sol = self._solve_mde_forward(mde, y0, t, dt)
+
+        dim = rv.mean.shape[0]
+        y_end = sol.y[:, -1]
+        new_mean = y_end[:dim]
+        new_cov_sqrt = y_end[dim:].reshape((dim, dim))
```

Comment:
this block of code exists for both forward implementations, right? Do you think it is worth trying to factor this into a common implementation, or do you reckon it is wasted efforts?

Reply:
I thought about this but there would be a naming issue since once we get the square-root covariance and once the covariance (in 251). If that's ok for you I can change it.

Reply:
Hmmm I dont know. I am a bit worried about all the flattening and slicing that is happening here, so in case some error comes up, it would be good to only have to fix it in one place. I see the naming though, and perhaps it is actually more readable if it is *not* factored out. As long as it only exists, like, twice, I am fine with a bit of duplication. 

Long story short, I dont know -- your call!

Reply:
Would something like `new_cov_or_cov_sqrt` resolve the naming issue?

Reply:
I think it would help, especially with an accompanying comment that explains why the `or`. Not sure whether it entirely resolves the issue though...